glib/src/gobject/binding_group.rs

```diff
@@ -140,19 +143,26 @@ impl<'a> BindingGroupBuilder<'a> {
             user_data: ffi::gpointer,
         ) -> ffi::gboolean {
             let transform_data =
-                &*(user_data as *const (TransformFn, TransformFn, ParamSpec, ParamSpec));
+                &*(user_data as *const (TransformFn, TransformFn, String, ParamSpec));
+            let binding = from_glib_borrow(binding);
 
             match (transform_data.1.as_ref().unwrap())(
-                &from_glib_borrow(binding),
+                &binding,
                 &*(from_value as *const Value),
             ) {
                 None => false,
                 Some(res) => {
+                    let pspec_name = transform_data.2.clone();
+                    let source = binding.source().unwrap();
```

Comment:
This is not thread-safe I think. The source might be unset while the closure is running or not? CC @jf2048

Reply:
Yes, it seems the real source is kept on the stack: https://gitlab.gnome.org/GNOME/glib/-/blob/91407cd7efec344f6bb5fd88fd7f771afb4fed25/gobject/gbinding.c

Reply:
This should be thread safe actually. `GBinding:source` is construct only, and the notify handler will always hold a reference while the transform function is running

Reply:
How is this construct-only if there's `binding_group.set_source(Some(&source))` (in the test)? :)

Reply:
This is in the `GBinding`, not the `GBindingGroup`. The binding group creates a new `GBinding` for each property, each `GBinding` holds a construct only weak ref to the source.

Reply:
Uh... right. Seems indeed fine then :)